leakcanary-android-core/src/androidTest/java/leakcanary/LeakActivityTest.kt

```diff
@@ -94,7 +93,28 @@ internal class LeakActivityTest {
       .check(matches(isDisplayed()))
   }
 
-  private fun insertHeapDump(block: HprofWriterHelper.() -> Unit) {
+  @Test
+  fun importHeapDumpFile() {
+    val latch = CountDownLatch(1)
+    LeakCanary.config = LeakCanary.config.copy(onHeapAnalyzedListener = {
+      DefaultOnHeapAnalyzedListener.create().onHeapAnalyzed(it)
+      latch.countDown()
+    })
```

Comment:
@pyricau Do you think we should be extra careful here and reset to the default `OnHeapAnalyzedListener`?
We could either do this in `@Before` and recreate a new `Config` or maybe in a JUnit rule?

Reply:
I do think we should reset it, yes, though I would do that at the end of the current test, and wrap the whole thing in a run / finally to make sure we reset it even on failures.

Reply:
Not sure about the best naming though.
I tried with `tryAndRestoreConfig(block: () -> Unit)`.